mod-audit-server/src/main/java/org/folio/verticle/acquisition/consumers/InvoiceLineEventsHandler.java

```diff
@@ -0,0 +1,62 @@
+package org.folio.verticle.acquisition.consumers;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import io.vertx.kafka.client.producer.KafkaHeader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.kafka.AsyncRecordHandler;
+import org.folio.kafka.KafkaHeaderUtils;
+import org.folio.kafka.exception.DuplicateEventException;
+import org.folio.rest.jaxrs.model.InvoiceLineAuditEvent;
+import org.folio.rest.util.OkapiConnectionParams;
+import org.folio.services.acquisition.InvoiceLineAuditEventsService;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class InvoiceLineEventsHandler implements AsyncRecordHandler<String, String> {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private final InvoiceLineAuditEventsService invoiceLineAuditEventsService;
+  private final Vertx vertx;
+
+  public InvoiceLineEventsHandler(Vertx vertx,
+                                  InvoiceLineAuditEventsService invoiceLineAuditEventsService) {
+    this.vertx = vertx;
+    this.invoiceLineAuditEventsService = invoiceLineAuditEventsService;
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> kafkaConsumerRecord) {
+    Promise<String> result = Promise.promise();
```

Comment:
I think it would be better to return future straight instead of using promise

Reply:
> I think it would be better to return future straight instead of using promise

This can be done in later PRs by extracting every consumer into an abstract class